### PR TITLE
Add Hero's Journey page to CMS

### PR DESCRIPTION
**Pull request by mwharrison**
API representation: 

```
{
    "id": 284,
    "meta": {
        "slug": "heros-journey",
        "seo_title": "",
        "search_description": "",
        "type": "pages.HeroJourneyPage",
        "detail_url": "http://openstax.org/api/v2/pages/284/",
        "html_url": "http://openstax.org/heros-journey/",
        "show_in_menus": false,
        "first_published_at": "2019-02-18T13:10:49.165596-06:00",
        "parent": {
            "id": 29,
            "meta": {
                "type": "pages.HomePage",
                "detail_url": "http://openstax.org/api/v2/pages/29/",
                "html_url": "http://openstax.org/"
            },
            "title": "Openstax Homepage"
        }
    },
    "title": "Hero's Journey",
    "books": [
        {
            "type": "heading",
            "value": "Get your Hero Badge",
            "id": "5ba55d48-713a-4cd7-b266-2dcd2c1f3822"
        },
        {
            "type": "subheading",
            "value": "You're only a few steps away from becoming an official OpenStax Hero....",
            "id": "98e835ba-485c-481a-b382-62aac150a152"
        },
        {
            "type": "description",
            "value": "Thanks to open alternatives like OpenStax, textbook prices have started to fall for the first time in 50 years. Instructors who choose affordable, open materials aren't just championing in their classroom -- they are causing a market-wide shift that's making school more affordable for all learners.",
            "id": "137c470d-bc4e-4c24-9cea-85d2434f5bad"
        },
        {
            "type": "book_heading",
            "value": "Look at the books",
            "id": "c357d78f-d5af-48b6-8e22-0ada2aa456fc"
        },
        {
            "type": "book_description",
            "value": "You already completed step one when you made an account. \r\nThe next step is to review the book to see if it could be useful in your course. \r\nTake a look, then come back here to continue your quest for the Hero Badge!",
            "id": "6f457f7c-a8a9-48ba-b6b3-9dab9d95c214"
        },
        {
            "type": "books_link",
            "value": "/subjects",
            "id": "9361b1ee-273d-46a9-aa7b-078a240492dc"
        },
        {
            "type": "books_link_text",
            "value": "Look at the books",
            "id": "163eaa84-1810-43a1-bb61-389a33f59576"
        }
    ],
    "quiz": [
        {
            "type": "heading",
            "value": "Pop Quiz!",
            "id": "1fc1b8ec-c7cc-409f-b936-70b586082ecb"
        },
        {
            "type": "skip_link",
            "value": "/adoption",
            "id": "a500d178-9818-489d-a0d2-0d58c5af2b5b"
        },
        {
            "type": "skip_link_text",
            "value": "Take me straight to the adoption form",
            "id": "848beced-abd5-4450-a8ac-9e0224701490"
        },
        {
            "type": "complete_message",
            "value": "You aced it!",
            "id": "e6843dd0-bfac-46ac-89e6-80f99baa2798"
        },
        {
            "type": "questions",
            "value": [
                {
                    "question": "All our work is devoted to helping students succeed because OpenStax is a(n) ___________",
                    "answers": [
                        {
                            "correct": false,
                            "text": "Urban legend"
                        },
                        {
                            "correct": true,
                            "text": "Non-profit organization dedicated to breaking down barriers to education"
                        },
                        {
                            "correct": false,
                            "text": "Friendly woodland creature"
                        },
                        {
                            "correct": false,
                            "text": "Natural sprinter"
                        }
                    ]
                },
                {
                    "question": "We have another question OpenStax is a(n) ___________",
                    "answers": [
                        {
                            "correct": false,
                            "text": "Something wrong"
                        },
                        {
                            "correct": false,
                            "text": "Another silly answer"
                        },
                        {
                            "correct": true,
                            "text": "What you should select"
                        },
                        {
                            "correct": false,
                            "text": "You have gone too far"
                        }
                    ]
                }
            ],
            "id": "96319aca-b50c-459f-9551-c306273cc05c"
        }
    ],
    "quiz_complete": [
        {
            "type": "heading",
            "value": "Now that you've aced all our quiz questions, it's time to try out OpenStax in your course!",
            "id": "0585bf20-999d-4f69-b9b0-1539f07c5671"
        },
        {
            "type": "description",
            "value": "If you've decided to include our book in your syllabus - whether it's the primary text, a supplemental resource, or just a few chapters - let us know.",
            "id": "26312d43-169f-4f2a-a1b8-8903062e1107"
        },
        {
            "type": "instructions",
            "value": "More stuff about how you should fill this out",
            "id": "949aff7b-d6a9-4dff-99db-15c71dd3fd5f"
        },
        {
            "type": "link_url",
            "value": "/adoption",
            "id": "1eed2f2b-8e30-4811-a4ff-c14358024186"
        },
        {
            "type": "link_text",
            "value": "I'm ready to try OpenStax",
            "id": "3ec6dd3e-0d2c-441f-acbc-655d3e852440"
        }
    ],
    "share": [
        {
            "type": "heading",
            "value": "Congratulations",
            "id": "b52ed0fc-925b-4c4a-9969-b3ae4f6e9908"
        },
        {
            "type": "description",
            "value": "You're a true hero, and we can't thank you enough for the work you do. There's just one more step before you get your official OpenStax Hero Badge: share your story with us.",
            "id": "564ae1c3-0e9d-4ccd-824c-c4220d9325d7"
        },
        {
            "type": "instructions",
            "value": "Tell us what you've learned, what you're most excited about for the future with your new text, or why you think it's important to share the power of an open textbook.",
            "id": "89711753-456d-4628-81fc-5c90683b0f8d"
        }
    ],
    "thanks": [
        {
            "type": "heading",
            "value": "Thank you, and welcome to the OpenStax family. Check your inbox for your Hero Badge!",
            "id": "7e0638c3-6bb5-4512-bbac-7deddad64f8a"
        }
    ]
}
```

### CODE REVIEW

**Comment by mwharrison** [comment]
@RoyEJohnson - per our conversation, I left the images out - and figured you would be getting the `first_name` from the accounts API.

File: pages/models.py [hunk 1]
```diff
@@ -2586,3 +2587,69 @@ class InstitutionalPartnership(Page):
...
+        ('link_url', blocks.CharBlock()),
+        ('link_text', blocks.CharBlock())
+    ], null=True)
+    share = StreamField([
+        ('heading', blocks.CharBlock()),
+        ('description', blocks.TextBlock()),
+        ('instructions', blocks.TextBlock()),
+    ], null=True)
+    thanks = StreamField([
+        ('heading', blocks.CharBlock()),
```
Comment (RoyEJohnson) [inline pages/models.py]: Sonya made heading pretty long, here - something like 185 characters, so you may need to change its definition, I don't konw. She also added a `description` (like all the other descriptions), so we might as well add that.

Comment (mwharrison) [reply pages/models.py]: I've changed this to a TextBlock and added a description field.

File: pages/models.py [hunk 2]
```diff
@@ -2586,3 +2587,69 @@ class InstitutionalPartnership(Page):
...
+
+class HeroJourneyPage(Page):
+    books = StreamField([
+        ('heading', blocks.CharBlock()),
+        ('subheading', blocks.CharBlock()),
+        ('description', blocks.TextBlock()),
+        ('book_heading', blocks.CharBlock()),
+        ('book_description', blocks.TextBlock()),
+        ('books_link', blocks.CharBlock()),
+        ('books_link_text', blocks.CharBlock()),
```
[the remark this answers is not in the document]
Comment (mwharrison) [reply pages/models.py]: Added!

**Pull request merged**